RealmSwift/Tests/ObjectCreationTests.swift

```diff
@@ -0,0 +1,340 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import XCTest
+import RealmSwift
+import Foundation
+
+class ObjectCreationTests: TestCase {
+
+    // MARK: Init tests
+    func testInitWithDefaults() {
+        // test all properties are defaults
+        let object = SwiftObject()
+        XCTAssertNil(object.realm)
+
+        // test defaults values
+        verifySwiftObjectWithDictionaryLiteral(object, dictionary: SwiftObject.defaultValues(), boolObjectValue: false, boolObjectListValues: [])
+
+        // test realm properties are nil for standalone
+        XCTAssertNil(object.realm)
+        XCTAssertNil(object.objectCol.realm)
+        XCTAssertNil(object.arrayCol.realm)
+    }
+
+    func testInitWithDictionary() {
+        // dictionary with all values specified
+        let valueCreator =  {
```

Comment:
This can be a lot simpler:

```swift
let valueCreator: NSDictionary = [
    "boolCol": true,
    "intCol": 1,
    "floatCol": 1.1,
    "doubleCol": 11.1,
    "stringCol": "b",
    "binaryCol": "b".dataUsingEncoding(NSUTF8StringEncoding)! as NSData,
    "dateCol": NSDate(timeIntervalSince1970: 2)
]
```